v3/newrelic/transaction.go

```diff
@@ -269,6 +271,76 @@ func (txn *Transaction) AcceptDistributedTraceHeaders(t TransportType, hdrs http
 	txn.thread.logAPIError(txn.thread.AcceptDistributedTraceHeaders(t, hdrs), "accept trace payload", nil)
 }
 
+//
+// AcceptDistributedTraceHeadersFromJSON works just like AcceptDistributedTraceHeaders(), except
+// that it takes the header data as a JSON string à la DistributedTraceHeadersFromJSON(). Additionally
+// (unlike AcceptDistributedTraceHeaders()) it returns an error if it was unable to successfully
+// convert the JSON string to http headers. There is no guarantee that the header data found in JSON
+// is correct beyond conforming to the expected types and syntax.
+//
+func (txn *Transaction) AcceptDistributedTraceHeadersFromJSON(t TransportType, jsondata string) error {
+	hdrs, err := DistributedTraceHeadersFromJSON(jsondata)
+	if err != nil {
+		return err
+	}
+	txn.AcceptDistributedTraceHeaders(t, hdrs)
+	return nil
+}
+
+//
+// DistributedTraceHeadersFromJSON takes a set of distributed trace headers as a JSON-encoded string
+// and emits a http.Header value suitable for passing on to the
+// txn.AcceptDistributedTraceHeaders() function.
+//
+// For example, given the input string
+//   `{"traceparent": "frob", "tracestate": "blorfl", "newrelic": "xyzzy"}`
+// This will emit an http.Header value with headers "traceparent", "tracestate", and "newrelic".
+//
+// This is a convenience function provided for cases where you receive the trace header data
+// already as a JSON string and want to avoid manually converting that to an http.Header.
+//
+// The JSON string must be a single object whose values may be strings or arrays of strings.
+// These are translated directly to http headers with singleton or multiple values.
+//
+func DistributedTraceHeadersFromJSON(jsondata string) (hdrs http.Header, err error) {
+	var raw interface{}
+	hdrs = http.Header{}
+	if jsondata == "" {
```

Comment:
Should we check for `nil` here, as well as an empty string? I'm still wrapping my head around what errors are likely to be made when calling this function.

Reply:
A string value can't be nil.